Build failure: n8n (on aarch64-linux)

### Steps To Reproduce

Steps to reproduce the behavior:
1. `nix build nixpkgs/6c0b7a92c30122196a761b440ac0d46d3d9954f1#n8n --impure`
2. Notice the build ends with `npm ERR! signal SIGILL`

### Build log

https://gist.github.com/gepbird/f575c87420fb835c7d1de6f376122a75

### Additional context

Builds fine on x86_64-linux

### Notify maintainers

@freezeboy @k900

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-linux"`
 - host os: `Linux 6.1.63, NixOS, 24.05 (Uakari), 24.05.20240517.4a6b83b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - channels(root): `"nixos-23.11.1115.5de0b32be6e8"`
 - nixpkgs: `/nix/store/0aavdx9m5ms1cj5pb1dx0brbrbigy8ij-source`

```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/networking/n8n/default.nix
+++ b/pkgs/applications/networking/n8n/default.nix
@@ -19,13 +19,17 @@ nodePackages.n8n.override {
     pkgs.postgresql
   ];
 
-  # Oracle's official package on npm is binary only (WHY?!) and doesn't provide binaries for aarch64.
-  # This can supposedly be fixed by building a custom copy of the module from source, but that's way
-  # too much complexity for a setup no one would ever actually run.
-  #
-  # NB: If you _are_ actually running n8n on Oracle on aarch64, feel free to submit a patch.
   preRebuild = lib.optionalString stdenv.isAarch64 ''
+    # Oracle's official package on npm is binary only (WHY?!) and doesn't provide binaries for aarch64.
+    # This can supposedly be fixed by building a custom copy of the module from source, but that's way
+    # too much complexity for a setup no one would ever actually run.
+    #
+    # NB: If you _are_ actually running n8n on Oracle on aarch64, feel free to submit a patch.
     rm -rf node_modules/oracledb
+
+    # This package tries to load a prebuilt binary for a different arch, and it doesn't provide the binary for aarch64.
+    # It is marked as an optional dependency in pnpm-lock.yaml and there are no other references to it n8n.
+    rm -rf node_modules/@sap/hana-client
   '';
 
   # makes libmongocrypt bindings not look for static libraries in completely wrong places

